sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -151,11 +151,14 @@ private void processNonCachedEvent(SentryEvent event) {
 
     PackageInfo packageInfo = getPackageInfo();
     if (packageInfo != null) {
+      String versionCode = getVersionCode(packageInfo);
+
       if (event.getRelease() == null) {
-        event.setRelease(packageInfo.packageName + "-" + packageInfo.versionName);
+        event.setRelease(
+            packageInfo.packageName + "@" + packageInfo.versionName + "+" + versionCode);
```

Comment:
`versionCode` is a `long`
 `versionName` is a `String`, and it could be a raw String, guess there's no `Regex` validator.
However, I've never seen something different than `x.x.x-something`
What should we do if we find a `@` or `+` on a `versionName`?

Reply:
Maybe rename to `-` to be safe? like `versionName.replace("@", "-")`?
If it's not common we can just live with it, as you prefer

Reply:
@bruno-garcia yep, for safety we can do this.
So we replace both reserved symbols, right?
@ to -
+ to - as well? Guess we need to use a different one here, so we could know how it was the original one.

Reply:
Sounds good. Your call.

Reply:
agreed with @HazAT to keep as it is, backend could get the last "@" and "+" and still parse it correctly